session.go

```diff
@@ -483,15 +500,38 @@ type DialInfo struct {
 	Username string
 	Password string
 
-	// PoolLimit defines the per-server socket pool limit. Defaults to 4096.
-	// See Session.SetPoolLimit for details.
+	// PoolLimit defines the per-server socket pool limit. Defaults to
+	// DefaultConnectionPoolSize. See Session.SetPoolLimit for details.
 	PoolLimit int
 
 	// PoolTimeout defines max time to wait for a connection to become available
-	// if the pool limit is reaqched. Defaults to zero, which means forever.
-	// See Session.SetPoolTimeout for details
+	// if the pool limit is reached. Defaults to zero, which means forever. See
+	// Session.SetPoolTimeout for details
 	PoolTimeout time.Duration
 
+	// ReadTimeout defines the maximum duration to wait for a response from
+	// MongoDB.
+	//
+	// This effectively limits the maximum query execution time. If a MongoDB
+	// query duration exceeds this timeout, the caller will receive a timeout,
+	// however MongoDB will continue processing the query. This duration must be
+	// large enough to allow MongoDB to execute the query, and the response be
+	// received over the network connection.
+	//
+	// Only limits the network read - does not include unmarshalling /
+	// processing of the response. Defaults to DialInfo.Timeout. If 0, no
+	// timeout is set.
+	ReadTimeout time.Duration
```

Comment:
The wording can be a bit misleading IMHO. What is happening is: 
1. if ReadTimeout > 0, use it
2 else if DialInfo.Timeout > 0 use it 
3. else no timeout is set, wait indefinitely ?

I would word it like this: 
```
// Only limits the network read - does not include unmarshalling /
// processing of the response. Defaults to DialInfo.Timeout. If , ReadTimeout is 0 and Timeout is 0 no
// timeout is set.
```

Reply:
I think your comment is missing words, but the flow you describe is what the existing comment says? I tested it on the DB guy.

Reply:
You are right. Ignore it please.